packages/dx-demo-shell/src/demo-viewer/source-code.jsx

```diff
@@ -34,16 +38,19 @@ export class SourceCode extends React.PureComponent {
       height: 'auto',
       gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
     });
-    this.foldBlockStartLines.forEach((lineNumber) => {
-      editor.foldCode(CodeMirror.Pos(lineNumber, 0));
-    });
-    this.importantLines.forEach((lineNumber) => {
-      editor.addLineClass(lineNumber, 'background', 'CodeMirror-important-line');
-    });
+    this.applySpecialCodeOptions();
   }
 
-  prepareSourceCode() {
-    const { themeName, sectionName, demoName } = this.props;
+  componentWillReceiveProps(nextProps) {
```

Comment:
It is [obsolete](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops).